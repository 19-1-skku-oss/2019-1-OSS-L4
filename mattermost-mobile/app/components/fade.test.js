// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Animated, Text} from 'react-native';
import {shallow} from 'enzyme';

import Fade from './fade';

jest.useFakeTimers();

describe('Fade', () => {
    const baseProps = {
        visible: true,
        disabled: true,
    };

    function getWrapper(props = {}) {
        const dummyText = 'text';

        return shallow(
            <Fade
                {...baseProps}
                {...props}
            >
                <Text>{dummyText}</Text>
            </Fade>
        );
    }

    test('should render {opacity: 1}', () => {
        const wrapper = getWrapper({visible: true});

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper).toHaveStyle('opacity', new Animated.Value(1));
    });

    test('should render {opacity: 0}', () => {
        const wrapper = getWrapper({visible: false});

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper).toHaveStyle('opacity', new Animated.Value(0));
    });

    test('should not change opacity when disabled flag is switched', () => {
        const wrapper = getWrapper({disabled: false});

        expect(wrapper).toHaveStyle('opacity', new Animated.Value(1));
        wrapper.setProps({disabled: true});
        expect(wrapper).toHaveStyle('opacity', new Animated.Value(1));
    });

    test('should not change opacity when props stay the same', () => {
        const wrapper = getWrapper({visible: true});

        expect(wrapper).toHaveStyle('opacity', new Animated.Value(1));
        wrapper.setProps({visible: true});
        expect(wrapper).toHaveStyle('opacity', new Animated.Value(1));
    });
});
