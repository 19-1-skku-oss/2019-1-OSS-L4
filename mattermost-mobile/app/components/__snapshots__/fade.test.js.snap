// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fade should render {opacity: 0} 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 0,
        },
      ],
    }
  }
>
  <Text>
    text
  </Text>
</AnimatedComponent>
`;

exports[`Fade should render {opacity: 1} 1`] = `
<AnimatedComponent
  style={
    Object {
      "opacity": 1,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
    }
  }
>
  <Text>
    text
  </Text>
</AnimatedComponent>
`;
